src/Sarif/NotYetAutoGenerated/ResultEqualityComparer.cs

```diff
@@ -0,0 +1,596 @@
+            {
+                if (left.WorkItemUris == null || right.WorkItemUris == null)
+                {
+                    return false;
+                }
+
+                if (left.WorkItemUris.Count != right.WorkItemUris.Count)
+                {
+                    return false;
+                }
+
+                for (int index_6 = 0; index_6 < left.WorkItemUris.Count; ++index_6)
+                {
+                    if (left.WorkItemUris[index_6] != right.WorkItemUris[index_6])
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            if (!object.ReferenceEquals(left.ConversionProvenance, right.ConversionProvenance))
+            {
+                if (left.ConversionProvenance == null || right.ConversionProvenance == null)
+                {
+                    return false;
+                }
+
+                if (left.ConversionProvenance.Count != right.ConversionProvenance.Count)
+                {
+                    return false;
+                }
+
+                for (int index_7 = 0; index_7 < left.ConversionProvenance.Count; ++index_7)
+                {
+                    if (!PhysicalLocation.ValueComparer.Equals(left.ConversionProvenance[index_7], right.ConversionProvenance[index_7]))
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            if (!object.ReferenceEquals(left.Fixes, right.Fixes))
+            {
+                if (left.Fixes == null || right.Fixes == null)
+                {
+                    return false;
+                }
+
+                if (left.Fixes.Count != right.Fixes.Count)
+                {
+                    return false;
+                }
+
+                for (int index_8 = 0; index_8 < left.Fixes.Count; ++index_8)
+                {
+                    if (!Fix.ValueComparer.Equals(left.Fixes[index_8], right.Fixes[index_8]))
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            if (!object.ReferenceEquals(left.Properties, right.Properties))
+            {
+                if (left.Properties == null || right.Properties == null || left.Properties.Count != right.Properties.Count)
+                {
+                    return false;
+                }
+
+                foreach (var value_6 in left.Properties)
+                {
+                    SerializedPropertyInfo value_7;
+                    if (!right.Properties.TryGetValue(value_6.Key, out value_7))
+                    {
+                        return false;
+                    }
+
+                    if (!object.Equals(value_6.Value, value_7))
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            // If we find a null correlation guid on either side, we will overwrite that null
```

Comment:
>  // [](start = 10, length = 4)

please note and comment on this behavior, which is non-obvious. if we compare two results, one of which has a correlation guid and the other which does not, we will flow the non-null correlation guid to the other result instance where it is null. the reason is that we've determined these two results are identical, and therefore the correlation guid relates to both.